lsm: rename lsm_batch_multiple to lsm_compaction_interval

matklad opened the pull request:
The first commit is a `sed` across the whole repository, the second one is me manually looking at the diff of the first commit and adjusting if necessary. I also grepped for `multiple`, which turned out two comments in constants.zig, also adjusted.

matklad commented on src/lsm/groove.zig:
```diff
@@ -117,7 +117,7 @@ comptime {
     comptime Storage: type,
     comptime Field: type,
-    comptime value_count_max: usize,
+    comptime table_value_count_max: usize,
```
This was a confusing name! This is not `value_count_max` for the entire tree, just for the table!

matklad commented on src/constants.zig:
```diff
@@ -53,9 +53,9 @@ comptime {
-    lsm_batch_multiple * stdx.div_ceil(pipeline_prepare_queue_max * 2, lsm_batch_multiple);
+pub const vsr_checkpoint_interval = journal_slot_count - lsm_compaction_interval *
+    // Trigger plus two pipelines, rounded up to `lsm_compaction_interval`.
+    (1 + stdx.div_ceil(pipeline_prepare_queue_max * 2, lsm_compaction_interval));
```
I am _almost_ motivated to introduce

```
stdx.round_up_to(value, multiple)
```

or

```
stdx.next_multiple(value, multiple)
```

jorangreef replied:
`next_multiple` is 🔥

matklad committed:
lsm: add clarity by naming table_value_count_max parameter

matklad committed:
vsr: simplify modulus computation

matklad committed:
lsm: rename lsm_batch_multiple part 1

Automatic rename with

   rg 'lsm_batch_multiple' --files-with-matches | xargs sed -i 's/lsm_batch_multiple/lsm_compaction_ops/g'

The new name, `lsm_compaction_ops`:

* matches `vsr_checkpoint_ops` (to be renamed in the future commit)
* is clear about the unit of measure (number of ops)
* is clear about this being compaction's bar

matklad committed:
lsm: rename lsm_batch_multiple part 2

Manual touch ups after automated rename

matklad committed:
vsr: rename vsr_checkpoint_interval

Automated change using

   rg 'vsr_checkpoint_interval' --files-with-matches | xargs sed -i 's/vsr_checkpoint_interval/vsr_checkpoint_ops/g'

The comments still talk about "checkpoint interval", but I think that's
correct: `vsr_checkpoint_ops` _is_ a checkpoint interval, but it is
named differently to put the units-of-measure (ops) as a suffix.

This makes the code of the form `op % vsr_checkpoint_ops` more natural.

sentientwaffle commented on src/constants.zig:
```diff
@@ -53,9 +53,9 @@ comptime {
-pub const vsr_checkpoint_interval = journal_slot_count -
-    lsm_batch_multiple -
-    lsm_batch_multiple * stdx.div_ceil(pipeline_prepare_queue_max * 2, lsm_batch_multiple);
+pub const var_checkpoint_ops = journal_slot_count -
```
**vsr**_checkpoint_ops, not **var**_checkpoint_ops

matklad replied:
Oh dear, that’s an excellent catch! 

I so clearly remember re-reading my sed invocation to make sure there are no typos 🤣

sentientwaffle commented on src/vsr.zig:
```diff
@@ -1563,8 +1563,8 @@ test "Checkpoint ops diagram" {
-        const last_slot = (op % constants.journal_slot_count) + 1 == constants.journal_slot_count;
+        const last_beat = (op + 1) % constants.lsm_batch_multiple == 0;
+        const last_slot = (op + 1) % constants.journal_slot_count == 0;
 
```
:heart:

sentientwaffle reviewed (approved):
The commit message still mentions `var_checkpoint_ops` but the code all looks correct :smile:

matklad commented:
TIL: rewording the commit message doesn't dismiss the review!
